Week-9/todoapplication/src/components/TodosList.js

```diff
@@ -0,0 +1,25 @@
+import React, { useContext } from "react";
+import "../AppWideCSS.css";
+import Todo from "./Todo";
+import TodoContext from "../store/todo-context";
+
+function TodosList(props) {
+  const ctx = useContext(TodoContext);
+  const todos = [...ctx.todosList];
+  console.log(todos);
+  const displayTodos = todos.map((todo) => {
+    return <Todo todo={todo} key={todo.id} />;
+  });
+  // console.log(displayTodos);
+
+  const fallbackUI = <span className="lighttext">Add some todos</span>;
+  return (
```

Comment:
if(!todos?.length) {
    return <span className="lighttext">Add some todos</span>
}

const renderTodo = (todo) => {
    return <Todo todo={todo} key={todo.id} />;
}

return (
    <div className="todolist">
        <h1>Todo List</h1>
         {todos.map(renderTodo)}
    </div>
)

Reply:
Understood your code separation. 
But I did a similar thing here and it looks neat.